arch/arm64/src/imx9/imx9_boot.c

```diff
@@ -44,19 +44,39 @@
 #include "imx9_gpio.h"
 #include "imx9_lowputc.h"
 
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC  1700000000
```

Comment:
Maybe this is better to be a CONFIG_ , if the correct CPU speed cannot be determined from elsewhere?

Reply:
Another option would be to make the CPU frequency as the kconfig variable and use that here. I think the systick timer runs at the same frequency as the CPU / Core does

Reply:
Yes, something to do still